stream_alert_cli/runner.py

```diff
@@ -141,24 +144,62 @@ def athena_handler(options):
                 list: 'array<string>'
             }
 
-            for key_name, key_type in sanitized_schema.iteritems():
-                # Transform the {} or [] into hashable types
-                if key_type == {}:
-                    key_type = dict
-                elif key_type == []:
-                    key_type = list
-
-                athena_schema[key_name] = schema_type_mapping[key_type]
-
-            schema_statement = ''.join(['{0} {1},'.format(key_name, key_type)
-                                        for key_name, key_type
-                                        in athena_schema.iteritems()])[:-1]
-            query = ('CREATE EXTERNAL TABLE {table_name} ({schema})'
-                     'PARTITIONED BY (dt string)'
-                     'ROW FORMAT SERDE \'org.openx.data.jsonserde.JsonSerDe\''
+            def add_to_athena_schema(schema, root_key=''):
+                """Helper function to add sanitized schemas to the Athena table schema"""
+                # Setup the root_key dict
+                special_key = None
+                if root_key and not athena_schema.get('root_key'):
+                    athena_schema[root_key] = {}
+
+                for key_name, key_type in schema.iteritems():
+                    # Transform the {} or [] into hashable types
+                    if key_type == {}:
+                        special_key = dict
+                    elif key_type == []:
+                        special_key = list
+
+                    # Account for envelope keys
+                    if root_key:
+                        if special_key:
+                            athena_schema[root_key][key_name] = schema_type_mapping[special_key]
+                        athena_schema[root_key][key_name] = schema_type_mapping[key_type]
+                    else:
+                        if special_key:
+                            athena_schema[key_name] = schema_type_mapping[special_key]
+                        athena_schema[key_name] = schema_type_mapping[key_type]
+
+            add_to_athena_schema(sanitized_schema)
+
+            # Support envelope keys
+            configuration_options = log_info.get('configuration')
+            if configuration_options:
+                envelope_keys = configuration_options.get('envelope_keys')
+                if envelope_keys:
+                    sanitized_envelope_keys = StreamAlert.sanitize_keys(envelope_keys)
+                    # Note: this key is wrapped in backticks to be Hive compliant
```

Comment:
Good note - why are back-ticks required for this key but they don't seem to be necessary for other keys?

Reply:
because of the special char in the key name

Reply:
Makes sense thanks for clarifying!